File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum, currSum = nums[0], 0
        for i in range(len(nums)):
            currSum += nums[i]
            if(currSum > maxSum):
                maxSum = currSum
            if(currSum < 0):
                currSum = 0
        return maxSum

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0], currSum = 0;
        for (int i=0;i<nums.length;i++){
            currSum += nums[i];
            if(maxSum<currSum){
                maxSum = currSum;
            } 
            if(currSum < 0){
                currSum = 0;
            }
        }
        return maxSum;
    }
}